minimint-api/src/module/http.rs

```diff
@@ -0,0 +1,39 @@
+use bytes::Bytes;
+use http_body::{Body as _, Full};
+
+pub type Body = Full<Bytes>;
+pub use http::*;
+
+pub struct Response(pub http::Response<Full<Bytes>>);
+impl Response {
+    pub fn json<T: serde::Serialize>(value: &T) -> serde_json::Result<Self> {
+        let bytes = serde_json::to_vec(value)?;
+        let body = Full::new(Bytes::from(bytes));
+        Ok(Self(http::Response::new(body)))
+    }
+
+    // read body as json, returns None if body was already read
+    pub fn body_json<T: serde::de::DeserializeOwned>(
+        &mut self,
+    ) -> Option<serde_json::Result<T>> {
+        // this doesn't block because full data is available instantly
+        let bytes = match futures::executor::block_on(self.0.body_mut().data()) {
+            Some(Ok(body)) => body,
+            None => return None,
+            Some(Err(infailable)) => match infailable {},
```

Comment:
typo: infallible, also what happens when this gets matched..? Will the `serde_json::from_slice(&bytes))` panic ?

Reply:
ok I guess will never be matched